From 35adfdffe7d2af552bc5c409e450b8fb62f195c8 Mon Sep 17 00:00:00 2001
From: QbProg <tholag@gmail.com>
Date: Sat, 19 May 2012 15:57:29 +0200
Subject: [PATCH] [warning-fix][missing-operators] When you define a class
 with a placement new operator without the corresponding
 delete one, the compiler complains. Also it complains when
 there's a const member without an assignment operator.

---
 inc/BRepMesh_Circ.hxx               |    4 ++++
 inc/BRepMesh_Vertex.hxx             |    4 ++++
 inc/BinLDrivers_DocumentSection.hxx |    4 ++++
 inc/Interface_FileParameter.hxx     |    4 ++++
 inc/gp_XY.hxx                       |    4 ++++
 inc/gp_XYZ.hxx                      |    4 ++++
 6 files changed, 24 insertions(+)

Index: oce/inc/BRepMesh_Circ.hxx
===================================================================
--- oce.orig/inc/BRepMesh_Circ.hxx
+++ oce/inc/BRepMesh_Circ.hxx
@@ -40,6 +40,10 @@ public:
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete (void * , void *)
+  {
+
+  }
 
   
   Standard_EXPORT   BRepMesh_Circ();
Index: oce/inc/BRepMesh_Vertex.hxx
===================================================================
--- oce.orig/inc/BRepMesh_Vertex.hxx
+++ oce/inc/BRepMesh_Vertex.hxx
@@ -47,6 +47,10 @@ public:
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete (void *,void *)
+  {
+
+  }
 
   
   Standard_EXPORT   BRepMesh_Vertex();
Index: oce/inc/BinLDrivers_DocumentSection.hxx
===================================================================
--- oce.orig/inc/BinLDrivers_DocumentSection.hxx
+++ oce/inc/BinLDrivers_DocumentSection.hxx
@@ -49,6 +49,10 @@ public:
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void * )
+  {
+
+  }
 
   //! Empty constructor <br>
   Standard_EXPORT   BinLDrivers_DocumentSection();
Index: oce/inc/Interface_FileParameter.hxx
===================================================================
--- oce.orig/inc/Interface_FileParameter.hxx
+++ oce/inc/Interface_FileParameter.hxx
@@ -52,6 +52,10 @@ public:
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete (void *,void *)
+  {
+
+  }
 
   
   Standard_EXPORT   Interface_FileParameter();
Index: oce/inc/gp_XY.hxx
===================================================================
--- oce.orig/inc/gp_XY.hxx
+++ oce/inc/gp_XY.hxx
@@ -57,6 +57,10 @@ public:
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete(void * , void *)
+  {
+
+  }
 
   //! Creates XY object with zero coordinates (0,0). <br>
       gp_XY();
Index: oce/inc/gp_XYZ.hxx
===================================================================
--- oce.orig/inc/gp_XYZ.hxx
+++ oce/inc/gp_XYZ.hxx
@@ -57,6 +57,10 @@ public:
   {
     if (anAddress) Standard::Free((Standard_Address&)anAddress); 
   }
+  void operator delete ( void * , void * )
+  {
+
+  }
 
   //! Creates an XYZ object with zero co-ordinates (0,0,0) <br>
       gp_XYZ();
